### ISSUE TITLE
Adding profile while dmg is mounted doesn't work

### ISSUE THREAD

**Issue by elferink** [Tunnelblick/Tunnelblick#270]
OSX 10.10.5
Tunnelblick 3.5.3 (build 4270.4371) (no previous installation of Tunnelblick on this mac)

After installing Tunnelblick I try to open a ovpn file, it asks if I want to open it with a program (Tunnelblick) I just downloaded (which is weird because I downloaded the installer and double clicked it).
After clicking confirm it doesn't add the profile.
All the while I had the dialog open that says to double click .ovpn files.

It looks like OSX tried to open the ovpn with the installer app in the mounted dmg.
I unmounted the dmg and tried again and it works as it should.

**Comment by jkbullard** [comment]
Yes, this happens sometimes, but is not reproducible.

It is an OS X problem really - the priority should be to use (A) the running program, then (B) the program in /Applications, and only after that other copies of the program.

The only thing I can think of that might help is if Tunnelblick ejects (unmounts?) the disk image after running the installer. I'll look into that.

**Comment by elferink** [comment]
Maybe the problem could easily be avoided by renaming the installer application to 'TunnelBlick Installer'?

**Comment by jkbullard** [comment]
Thanks for the suggestion, but I'm afraid that won't work. OS X doesn't use than name of the application for this purpose, it uses an entry in the application's Info.plist (a file contained within the application).

Tunnelblick's "installer" is just a copy of the application itself. That was originally done for simplicity (I think). So both the installer and the application contain the same Info.plist. Because the installer/application are digitally signed, they can't be altered, so the installer can't add the entry to the file when the program is installed.

It might be possible to use a true OS X Installer "package", but it would also contain a copy of the application, so I think it would have the same problem.

**Comment by elferink** [comment]
If you use a .pkg I would think there won't be 2 available .app's because one would be contained in the .pkg would it not? A .dmg mounts, but a .pkg doesn't (at least not visibly).

**Comment by jkbullard** [comment]
Good point, thanks. The problem is that the .app exists in the file system (on the .dmg). So it might be that the mere existence of the .app on the filesystem (in the package) would trigger it. Or maybe .apps are compressed or something in a .pkg, which would mean it doesn't really exist on the file system. I don't know much about the .pkg installer.

It would be a lot more work to make the installer a .pkg than to just eject the .dmg, and this problem seems to only happen on certain computers (not related to OS version as far as I can tell), and once the program is installed the .dmg is no longer needed* so I'm inclined not to do that.

* I have to double-check that, because the .dmg can contain configurations that are automatically installed, and I'm not sure when that happens: before, during, or after the .app is installed into /Applications.

**Comment by elferink** [comment]
Alright, let me know if you want me to test anything or want more info.

**Comment by jkbullard** [comment]
Tunnelblick 3.6beta09 has been released and ejects the disk image if it is mounted when Tunnelblick launches. This is done at the end of launching Tunnelblick if it has already been installed.

**Comment by jkbullard** [comment]
Sorry to be so late getting back to you about this.

It doesn't solve the problem completely, but Tunnelblick 3.6beta10 unmounts
the disk image when it launches normally (from /Applications). So if you
install Tunnelblick from the disk image (which means that it is running
from the disk image), and then click "Launch", Tunnelblick will launch,
unmount the disk image, and after that double-clicking a configuration
should work properly.

The problem with this is that if you don't click "Launch", but click "Quit"
instead, the unmount doesn't take place and the problem isn't solved, so I
am considering always launching Tunnelblick at the end of an install and
not providing the option to quit.

**Comment by elferink** [comment]
First, thank you for fixing this.
Second, I think you could get away with launching the application after installing.
I think it's not common to install something you're not going to use at that moment.
And the chances are someone will otherwise launch TunnelBlick by clicking a .opvn or .tblk and thus will be having the same issue.
